hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -157,10 +157,10 @@ public HoodieRecord truncateRecordKey(Schema recordSchema, Properties props, Str
 
   @Override
   public boolean isDelete(Schema recordSchema, Properties props) throws IOException {
-    if (this.data instanceof BaseAvroPayload) {
-      return ((BaseAvroPayload) this.data).isDeleted(recordSchema, props);
-    } else {
+    if (this.data instanceof AWSDmsAvroPayload || !(this.data instanceof BaseAvroPayload)) {
```

Comment:
The essential logic looks good to me.  Could you make the change so it's similar to the code before:
```
if (!(this.data instanceof AWSDmsAvroPayload) || this.data instanceof BaseAvroPayload) {
  return ((BaseAvroPayload) this.data).isDeleted(recordSchema, props);
} else {
  return !this.data.getInsertValue(recordSchema, props).isPresent();
}
```

Reply:
Updated the logic as per suggestion. It should be AND logic instead of OR. Please review.